packages/runtime-html/src/dom.ts

```
@@ -430,7 +430,9 @@ export class FragmentNodeSequence implements INodeSequence {
       }
     } else {
       this.isMounted = true;
-      parent.appendChild(this.fragment);
+      if (!parent.contains(this.fragment)) {
```

can we have some comment on which cases we need to perform this check? I'm unable to guess this.

---

For enhance the `host` and `fragment` will be same. However, instead of equality check I think that the more general `contains` check makes more sense in this case, as we might generally want to skip the append action if the host already has the child.

---

The idea is good but I'd prefer then if we throw instead, otherwise people try to do something, they don't know they're doing anything wrong, and nothing happens and they won't know why. Silent failures are the worst.
But if I'm not mistaken, `parent.appendChild` already throws if it's contained in the tree, so why not just let the DOM api throw? Is there a specific use case where we need to allow this?

---

Actually it does throw. We need this check only for enahnce. I felt propagating the enhance flag all the way to dom bit of an overkill. That's why I felt this check to be a reliable proxy. But the changed version looks much cleaner.